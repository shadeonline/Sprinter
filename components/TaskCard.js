import React, { useState } from 'react';
import { TouchableOpacity, StyleSheet, Text, View } from 'react-native';


const TaskCard = ({ task, onPress }) => {
    return (
        <TouchableOpacity
            style={styles.taskContainer}
            onPress={onPress} // Use the onPress prop here
        >
            <View style={styles.taskHeader}>
                <Text style={styles.taskTitle}>{task.taskTitle}</Text>
                <View style={styles.storyPointContainer}>
                    <Text style={styles.storyPoint}>{task.storyPoint ? task.storyPoint : '-'}</Text>
                </View>
            </View>
            <Text style={styles.deadline}>Deadline: {task.deadline}</Text>
        </TouchableOpacity>
    );
};

const styles = StyleSheet.create({
    taskContainer: {
        backgroundColor: 'white',
        padding: 16,
        margin: 16,
        borderRadius: 8,
        elevation: 2,
    },
    taskHeader: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        marginBottom: 8,
    },
    taskTitle: {
        fontSize: 16,
        fontWeight: 'bold',
    },
    storyPointContainer: {
        fontSize: 14,
        color: 'gray',
        backgroundColor: '#DCE0E8',
        paddingHorizontal: 20,
        paddingVertical: 2,
        borderRadius: 10,
    },
    storyPoint: {
        fontSize: 14,
        color: 'gray',
    },
    deadline: {
        fontSize: 14,
        color: 'gray',
    },
});

export default TaskCard;
