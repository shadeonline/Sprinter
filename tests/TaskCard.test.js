// TaskCard.test.js

import React from 'react';
import renderer from 'react-test-renderer';
import TaskCard from '../components/TaskCard'; // Adjust the path based on your project structure
import { useNavigation } from '@react-navigation/core';

// Mock React Navigation
jest.mock('@react-navigation/core', () => ({
  useNavigation: jest.fn(),
}));

describe('TaskCard', () => {
  it('matches snapshot', () => {
    // Mock a task object for testing
    const task = {
      taskTitle: 'Sample Task',
      storyPoint: '3',
      deadline: '30/09/23',
      status: 'Backlog',
    };

    // Mock the useNavigation function
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    const tree = renderer.create(<TaskCard task={task} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
