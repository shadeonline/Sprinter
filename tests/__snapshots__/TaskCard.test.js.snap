// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCard matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "white",
      "borderColor": "gray",
      "borderRadius": 8,
      "borderWidth": 3,
      "elevation": 2,
      "margin": 16,
      "opacity": 1,
      "padding": 16,
      "width": "85%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 8,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Sample Task
    </Text>
    <View
      style={
        {
          "backgroundColor": "#DCE0E8",
          "borderRadius": 10,
          "color": "gray",
          "fontSize": 14,
          "paddingHorizontal": 20,
          "paddingVertical": 2,
        }
      }
    >
      <Text
        style={
          {
            "color": "gray",
            "fontSize": 14,
          }
        }
      >
        3
      </Text>
    </View>
  </View>
  <Text
    style={
      {
        "color": "gray",
        "fontSize": 14,
      }
    }
  >
    Deadline: 
    30/09/23
  </Text>
</View>
`;
